tlatools/org.lamport.tlatools/src/tlc2/TraceExplorer.java

```diff
@@ -139,6 +143,21 @@ public static void writeSpecTEStreams(
 
 		final SpecTraceExpressionWriter writer = new SpecTraceExpressionWriter();
 
+		// Add all the model values as constants, they have to be reified (to mean something) in a TLA module.
```

Comment:
A few notes:
* Can/Should this move the SpecTraceExpressionWriter?  
* Where in the trace spec should this go?  A user doesn't care about the model values.
* How prevalent are specs that don't explicitly define model values?  IIRC the Toolbox has always explicitly generated model values.
* If this gets added, there should be a detailed comment that explains the Whys.

Reply:
> Can/Should this move the SpecTraceExpressionWriter?

I guess so, I didn't move because I have seen that other places were using the `SpecTraceExpressionWriter`, so I would have to maybe test the Toolbox to see the impact there (I plan to do a checkpoint there after changing something in the Expression Writer).

> Where in the trace spec should this go? A user doesn't care about the model values.

Got it, I'll add better comments if we use this.

> How prevalent are specs that don't explicitly define model values? IIRC the Toolbox has always explicitly generated 
model values.

I just took your own example, as you are a proficient user of TLA+, I have assumed that this would happen, but if this is something that you think does not happen enough, I would be satisfied not dealing with it now.

> If this gets added, there should be a detailed comment that explains the Whys.

Hunrum, yes, I am willing to add improved comments and more comments for the non-obvious code.

Reply:
Does your code call ModelValue#init?  If it does, please ignore. If not, what code calls it and why doesn't it call ModelValue#setValues?

Reply:
I'm calling setValues in line 148, but I'm not calling init, didn't need. But if it's preferable, I can call init too.